doc/release/1.12.0-notes.rst

```diff
@@ -23,6 +23,20 @@ Future Changes
 Compatibility notes
 ===================
 
+DeprecationWarning to error
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Indexing with floats raises IndexError,
+  e.g., a[0, 0.0].
+* Indexing with non-integer array_like raises ``IndexError``,
+  e.g., ``a['1', '2']``
+* Indexing with multiple ellipsis raises ``IndexError``,
+  e.g., ``a[..., ...]``.
+* Indexing with boolean where integer expected raises ``IndexError``,
+  e.g., ``a[False:True:True]``.
```

Comment:
Sure, go ahead. Just because I checked it anyway, maybe the "arrays" in there is confusing, hmm:
```
In [7]: np.arange(10)[True, ...]
IndexError: in the future, 0-d boolean arrays will be interpreted as a valid boolean index
```

Anyway, not for you to bother about right now, just do it any way you like, we have a 1.11 priority :).